### PR TITLE
CLOUDSTACK-8968: UI icon over VM snapshot to deploy user instance

### PR DESCRIPTION
**Pull request by nitin-maharana**
Added a new Icon in Instance page to launch the VM from the snapshot.

A new icon over VM snapshot object, which upon invoked will open a deploy user instance wizard, where there is no choice to choose zone and ISO or template to deploy the user instance from.

1) A new icon to indicate deploy user instance from VM snapshot
2) The new icon is placed as an operation over each VM snapshot object.
3) Clicking this icon will show the wizard to deploy user instance, which is same as wizard for normal user instance except that zone selection page and ISO/template selection page would not be present in this wizard.

### CODE REVIEW

**Comment by nitin-maharana** [comment]
Reference PR #1150 (Against 4.6). This PR (against 4.7) contains the same code change with one conflict which I resolved.

File: ui/scripts/instanceWizard.js [hunk 1]
```diff
@@ -294,53 +295,67 @@
...
                             }
                         }
-                    }
-                    if (selectedTemplateObj == null) {
-                        if (communityTemplateObjs != null && communityTemplateObjs.length > 0) {
-                            for (var i = 0; i < communityTemplateObjs.length; i++) {
-                                if (communityTemplateObjs[i].id == args.currentData.templateid) {
-                                    selectedTemplateObj = communityTemplateObjs[i];
-                                    break;
+                        if (selectedTemplateObj == null) {
```
Comment (rodrigo93) [inline ui/scripts/instanceWizard.js]: Hi @nitin-maharana 
Couldn't this _if_ and the following _ifs_ be merged into one? 
Like:

> if ( selectedTemplateObj == null && communityTemplateObjs != null && communityTemplateObjs.length > 0)

**Comment by swill** [comment]
@nitin-maharana please rebase as we currently have merge conflicts.  Thanks...

**Comment by rhtyd** [comment]
@nitin-maharana please rebase and share screenshots of what this fixes

A manual UI testing would be needed

**Comment by rhtyd** [comment]
@nitin-maharana can you rebase and change PR base to 4.8+ branches. Thanks.

**Comment by nitin-maharana** [comment]
@rhtyd: This is dependent on #1664, Once #1664 gets merged, I will make a new PR.

**Comment by rhtyd** [comment]
@nitin-maharana the feature is related, and exclusive; in that case please move your changes to PR #1664

**Comment by nitin-maharana** [comment]
@rhtyd : I discussed with @sateesh-chodapuneedi regarding this. I will give my changes to him. He will include my commit in his PR. Thanks.

**Comment by rhtyd** [comment]
Thanks @nitin-maharana

**Comment by nitin-maharana** [comment]
This change will come with #1664. Hence closing this.

**Comment by nitin-maharana** [comment]
Hi @rhtyd, I observed there is a change in the way we were maintaining the dictionary. The messages*.properties and dictioanry*.jsp file was removed and the content was moved to the l10n folder. How the change in l10n files reflect in the browser. Do we need to modify any other file to reflect the change?

**Comment by rhtyd** [comment]
@nitin-maharana yes, the master branch is JSP free now. The UI is purely JS/html/css based.
Yes, you'll need to add new keys only to the l10n (localisation) specific files and use them in the UI/JS code.
This also fixes the issue of adding the same set of keys to messages.properties file and then the dictionary/jsp files. The workflow is much easier now.

**Pull request closed without merging**